build.rs

```diff
@@ -41,71 +45,199 @@ fn wast_processor(out: &mut Testsuite, p: PathBuf) -> Option<Test> {
     })
 }
 
+/// Given a Testsuite and a path, process the path in case is a Emscripten
+/// wasm file.
+fn emscripten_processor(out: &mut Testsuite, p: PathBuf) -> Option<Test> {
+    let ext = p.extension()?;
+    // Only look at wast files.
+    if ext != "wasm" {
+        return None;
+    }
+
+    let outfile = {
+        let mut out_ext = p.clone();
+        out_ext.set_extension("out");
+        // let mut txt_ext = p.clone();
+        // txt_ext.set_extension("txt");
+        if out_ext.exists() {
+            out_ext
+        }
+        // else if txt_ext.exists() {
+        //     txt_ext
+        // }
+        else {
+            return None;
+        }
+    };
+
+    let testname = extract_name(&p);
+    let compiler = out.path.get(0).unwrap();
+    let body = format!(
+        "crate::run_emscripten(r#\"{}\"#, r#\"{}\"#, \"{}\")",
+        p.display(),
+        outfile.display(),
+        compiler
+    );
+
+    Some(Test {
+        name: testname.to_string(),
+        body: body.to_string(),
+    })
+}
+
+/// Given a Testsuite and a path, process the path in case is a WASI
+/// wasm file.
+fn wasi_processor(out: &mut Testsuite, p: PathBuf) -> Option<Test> {
+    let ext = p.extension()?;
+    // Only look at wast files.
+    if ext != "wasm" {
+        return None;
+    }
+
+    let outfile = {
+        let mut out_ext = p.clone();
+        out_ext.set_extension("out");
+        // let mut txt_ext = p.clone();
+        // txt_ext.set_extension("txt");
+        if out_ext.exists() {
+            out_ext
+        }
+        // else if txt_ext.exists() {
+        //     txt_ext
+        // }
+        else {
+            return None;
+        }
+    };
+
+    let testname = extract_name(&p);
+    let compiler = out.path.get(0).unwrap();
+    let body = format!(
+        "crate::run_wasi(r#\"{}\"#, r#\"{}\"#, \"{}\")",
+        p.display(),
+        outfile.display(),
+        compiler
+    );
+
+    Some(Test {
+        name: testname.to_string(),
+        body: body.to_string(),
+    })
+}
+
+fn is_truthy_env(name: &str) -> bool {
+    env::var(name).unwrap_or("0".to_string()) == "1"
```

Comment:
⛳ 
`env::var(name).ok().as_deref().unwrap_or("0") == "1"` avoids the extra allocation so does `env::var(name).map(|n| n == "1").unwrap_or_default()`

Alternatively `env::var(name).unwrap_or_else(|| "0".to_string()) == "1"` avoids the extra allocation on the happy path but not the unhappy path.

Super minor detail that doesn't matter here -- this is more of a general style note.

Reply:
```suggestion
env::var(name).map(|n| n == "1").unwrap_or_default()
```